app/models/service_template/copy.rb

```diff
@@ -0,0 +1,17 @@
+module ServiceTemplate::Copy
+  extend ActiveSupport::Concern
+
+  def template_copy(new_name = "Copy of " + name + Time.zone.now.to_s)
+    if template_valid? && type != 'ServiceTemplateAnsiblePlaybook'
+      ActiveRecord::Base.transaction do
+        dup.tap do |template|
```

Comment:
I had a concern about some attributes being duped, but @d-m-u assures me, it's been thoroughly considered and shouldn't be a problem.

Reply:
Same here, she was very reassuring.  😄 Plus I looked into the doc for `dup` which helped.  👍

Reply:
Was thinking more about this and wondering if we should override the `dup` method and have it also the `guid` value to `nil`.  This really seems like the pattern we want.

Thoughts?

Reply:
@gmcculloug I agree, that sounds like it belongs in the `UuidMixin`.

Reply:
@bdunne but you can just override initialize_copy like this's doing now, can this please get merged as first step to fix bug and if necessary i redo it later?

Reply:
@bdunne I like the idea of moving to the mixin for `guid` (in a separate PR), but does that limit using this pattern if additional fields need to be cleared on other models?

Reply:
> @bdunne I like the idea of moving to the mixin for `guid` (in a separate PR), but does that limit using this pattern if additional fields need to be cleared on other models?

I think it would just work.  `initialize_copy` in the uuid mixin to set the guid and call super should only affect types that include UuidMixin.  Either way, for another PR.